### PR TITLE
Fix build source tarball and enable CI for Red Hat Offline build

### PR DESCRIPTION
**Pull request by dseefeld**
Changes to enable building of the tarball from source-build and setting up CI for Offline build.  Offline build is building through corefx for now until other repos can get fixed.

Fixes dotnet/source-build#191

### CODE REVIEW

**Comment by dseefeld** [comment]
@dotnet-bot test CI please

**Comment by dseefeld** [comment]
@dotnet-bot  test CI please

**Comment by dseefeld** [comment]
@dotnet-bot test CI please

**Comment by dseefeld** [comment]
@dotnet-bot test CI please

**Comment by dseefeld** [comment]
@dotnet-bot test CI please

**Comment by dseefeld** [comment]
test CI please

**Comment by dseefeld** [comment]
@dotnet-bot test CI please

**Comment by dseefeld** [comment]
@dotnet-bot test CI please

File: netci.groovy [hunk 1]
```diff
@@ -59,6 +59,47 @@ def addPushJob(String project, String branch, String os, String configuration)
 };
 
 [true, false].each { isPR ->
+  ["RHEL7.2"].each { os->
+    ["Release", "Debug"].each { configuration ->
+
+      def shortJobName = "${os}_Offline_${configuration}";
+      def contextString = "${os} ${configuration}";
```
Comment (crummel) [inline netci.groovy]: Is this meant to be the same as the non-offline build trigger phrases?  As is I think "test RHEL7.2 Release" kicks both the offline and online builds.

File: src/reference-assemblies/Directory.Build.props [hunk 2]
```diff
@@ -1,10 +1,13 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+    <PropertyGroup>
+        <DisableBuildToolsRoslynVersion>true</DisableBuildToolsRoslynVersion>
```
Comment (crummel) [inline src/reference-assemblies/Directory.Build.props]: Is this related to the tarball or just a general build fix?

File: netci.groovy (hunk 1 above)
Comment (dseefeld) [reply netci.groovy]: No.  It should include Offline.  I'll change it to "${os} Offline ${configuration}"

File: src/reference-assemblies/Directory.Build.props (hunk 2 above)
Comment (dseefeld) [reply src/reference-assemblies/Directory.Build.props]: It applies to reference-assemblies, which are only built in the Offline build.

**Comment by dseefeld** [comment]
@dotnet-bot test CI please

File: src/reference-assemblies/Directory.Build.props (hunk 2 above)
Comment (weshaggard) [reply src/reference-assemblies/Directory.Build.props]: Is it causing a particular problem while building the ref-assemblies? If so can you give a note in a comment here of why it is disabled.

File: netci.groovy [hunk 3]
```diff
@@ -59,6 +59,47 @@ def addPushJob(String project, String branch, String os, String configuration)
...
+    ["Release", "Debug"].each { configuration ->
+
+      def shortJobName = "${os}_Offline_${configuration}";
+      def contextString = "${os} Offline ${configuration}";
+      def triggerPhrase = "(?i).*test\\W+${contextString}.*";
+
+      def newJob = job(Utilities.getFullJobName(project, shortJobName, isPR)){
+        steps{
+            shell("cd ./source-build;git submodule update --init --recursive");
+            shell("cd ./source-build;./build.sh /p:ArchiveDownloadedPackages=true /p:Configuration=${configuration} ${loggingOptions}");
```
Comment (weshaggard) [inline netci.groovy]: Don't we need to pass OfflineBuild=true here?

**Comment by weshaggard** [comment]
Changes look reasonable to me.

File: netci.groovy (hunk 3 above)
Comment (dseefeld) [reply netci.groovy]: No.  Not on this initial build.  OfflineBuild=true is set in the next statement: ./tarball-output/build.sh

**Pull request merged**